include/oneapi/tbb/concurrent_queue.h

```diff
@@ -131,8 +134,54 @@ class concurrent_queue {
         queue_allocator_traits::destroy(my_allocator, my_queue_representation);
         r1::cache_aligned_deallocate(my_queue_representation);
     }
+  
+    concurrent_queue& operator=( const concurrent_queue& other ){
+        //TODO: implement support for std::allocator_traits::propogate_on_container_copy_assignment
+        if (my_queue_representation != other.my_queue_representation) {
+            clear();
+	    my_queue_representation->assign(*other.my_queue_representation, my_allocator, copy_construct_item);
+	}
+	return *this;
+    }
+
+    concurrent_queue& operator=( concurrent_queue&& other ){
+        //TODO: implement support for std::allocator_traits::propogate_on_container_move_assignment
+        if (my_allocator == other.my_allocator){
+	    clear();
+	    my_queue_representation = other.my_queue_representation;
+	}
+	else{
+	    clear();
+            my_queue_representation->assign(*other.my_queue_representation, my_allocator, move_construct_item);
+	    other.clear();
```

Comment:
I thought one more time about this `clear` and it seems like it should not be done here, because we definitely need to move-assign allocators before doing the `my_queue_representation->assign` to re-allocate everything using `other.my_allocator`. In this case, `other.clear()` can result in an issue because `other.my_allocator` would be moved-from.

May be we can call `assign` method with `other.my_allocator`, then call `other.clear()` and only after that call allocators move assignment? Could you please check it using `container_move_support` tests?